### ISSUE TITLE
Somewhere between 2.2.25 and 2.2.30 opening a directory became slow to the point of failure. 

### ISSUE THREAD

**Issue by thegarbz** [haiwen/seadroid#900]
I have "recently" been experiencing problems browsing directories on the Android app. The app has been slow for a long time but I it didn't really become a problem until I tried accessing folders with more than ~1000-2000 files. The folder never opens just shows the spinning loading symbol forever. 

I started by downgrading to 2.2.30 but the problem persisted. 
I then downgraded to 2.2.25 and the problem is completely eliminated. Folders even with 7000+ files open close to instantly showing a loading icon for less than 1 second, and the entire traversing process is completely smooth. 

Somewhere between 2.2.25 and 2.2.30 the display of a director became so slow as to be effectively broken. This issue is present in all versions above 2.2.30 including 2.2.39 and persisted across a factory reset of the android device. 

Current Seafile server version 8.0.5

**Comment by maxiaoping** [comment]
Changes to the image cache system

**Comment by thegarbz** [comment]
Can this be fixed? It makes the app fail to do it's one task. I mean I'm happy sitting perpetually at 2.2.25 and thank you guys for making past versions available to roll back to, but that's hardly a neat solution :-)

**Comment by maxiaoping** [comment]
This issue will be fixed in the next release

**Comment by maxiaoping** [comment]
Please update the latest version 2.2.41

**Comment by thegarbz** [comment]
@maxiaoping Many thanks for fixing this. I just confirmed that it not only works, but performance has been improved over baseline:

Folder A 6478 files, Folder B 7509 files.

Seafile 2.2.25:
A 2.56 seconds
B 2.30 seconds

Seafile 2.2.39:
A -aborted- after 1minute. B unable to check since it was a subfolder of A.

Seafile 2.2.41:
A 1.86 seconds
B 1.64 seconds

Thanks for your action on this, especially since this issue meant I had to disable automatic updating on the Play Store.
